joomla/joomla-cms
# Make language flags in backend optional

Currently in a multilingual site the backend indicates the language in several places using a country flag. Since that is not always an appropriate thing some expressed the wish to hide those flags which is currently not possible.
Based on a discussion in https://github.com/joomla/joomla-cms/pull/12034 and based on some code started by @andrepereiradasilva this is what I came up with.
### Summary of Changes

Basically this adds the possibility to assign no image to a content language. Until now, a flag had to be chosen.
To avoid showing invalid images all over the place, the code had to be adjusted to check the value of the content language image property and fall back to a label or nothing if it wasn't specified.
### Testing Instructions
1. Set up a multilingual site, make sure you have some associations set as well
2. Adjust at least one content language so it doesn't have an image assigned. Leave at least one language with a flag so you can test both at the same time :smile: 
3. Check the various places where a flag is usually shown;
- Menu module shows a flag for a menu with a language specific home
- Menu manager shows the language specific home button as flag
- Template manager shows the language specific default button as flag
- Several lists show the language with a preceding flag
- Associations have a flag in the tooltip
- In a module edit, the "Menu Assignment" tab shows the language specific home menu items with a flag

A note: This is strictly about backend. As far as I know in frontend the only place the flags are used is the language switcher module, and that one already has a switch to toggle between flags and text.
### Documentation Changes Required

None that I'm aware of.
